BDArmory/Weapons/Missiles/MissileLauncher.cs

```diff
@@ -706,47 +706,41 @@ public override void OnStart(StartState state)
             }
 
-            IEnumerator<PartModule> partModules = part.Modules.GetEnumerator();
-            while (partModules.MoveNext())
+            warheadType = WarheadTypes.Kinetic; // Default to Kinetic if no appropriate modules are found.
+            foreach (var partModule in part.Modules)
             {
-                if (partModules.Current == null) continue;
-                if (partModules.Current.moduleName == "BDExplosivePart")
+                if (partModule == null) continue;
+                switch (partModule.moduleName)
                 {
-                    ((BDExplosivePart)partModules.Current).ParseWarheadType();
-                    if (((BDExplosivePart)partModules.Current).warheadReportingName == "Continuous Rod")
-                    {
-                        warheadType = WarheadTypes.ContinuousRod;
-                    }
-                    else
-                    {
-                        if (((BDExplosivePart)partModules.Current).tntMass > 0)
-                            warheadType = WarheadTypes.Standard;
+                    case "BDExplosivePart":
+                        ((BDExplosivePart)partModule).ParseWarheadType();
+                        if (((BDExplosivePart)partModule).warheadReportingName == "Continuous Rod")
+                            warheadType = WarheadTypes.ContinuousRod;
                         else
-                            warheadType = WarheadTypes.Kinetic;
-                    }
-                }
-                if (partModules.Current.moduleName == "ClusterBomb")
-                {
-                    clusterbomb = ((ClusterBomb)partModules.Current).submunitions.Count;
-                }
-                if (partModules.Current.moduleName == "MultiMissileLauncher" && weaponClass == WeaponClasses.Bomb)
-                {
-                    clusterbomb *= (int)((MultiMissileLauncher)partModules.Current).salvoSize;
-                }
-                if (partModules.Current.moduleName == "ModuleEMP")
-                {
-                    warheadType = WarheadTypes.EMP;
-                    StandOffDistance = ((ModuleEMP)partModules.Current).proximity;
-                }
-                if (partModules.Current.moduleName == "BDModuleNuke")
-                {
-                    warheadType = WarheadTypes.Nuke;
-                    StandOffDistance = BDAMath.Sqrt(((BDModuleNuke)partModules.Current).yield) * 500;
+                            warheadType = WarheadTypes.Standard;
+                        break;
+                    case "ClusterBomb":
+                        clusterbomb = ((ClusterBomb)partModule).submunitions.Count;
+                        break;
+                    case "MultiMissileLauncher":
+                        if (weaponClass == WeaponClasses.Bomb)
+                            clusterbomb *= (int)((MultiMissileLauncher)partModule).salvoSize;
+                        else continue;
```

Comment:
One issue with this is that MML launchers (such as the `VLSQuad` and reloadable rails) end up classified as Kinetic, since they have MML and MMRearm modules, but no warhead module.
Adding a `WarheadTypes.Launcher` type and switching this to
```csharp
else
{
    if (warheadType == WarheadTypes.Kinetic) warheadType = WarheadTypes.Launcher;
    continue; // Set the warhead type as launcher, but continue looking for other types.
}
```